src/ProjectService.Business/Commands/Task/FindTasksCommand.cs

```diff
@@ -10,113 +18,122 @@
+          return response.Message.Body;
+        }
 
-            try
-            {
-                var response = _requestClient.GetResponse<IOperationResult<IGetUsersDataResponse>>(
-                    IGetUsersDataRequest.CreateObj(userId)).Result;
+        errors.AddRange(response.Message.Errors);
 
-                if (response.Message.IsSuccess)
-                {
-                    return response.Message.Body;
-                }
+        _logger.LogWarning("Can not find user data with this id {UserId}: " +
+          $"{Environment.NewLine}{string.Join('\n', response.Message.Errors)}", userId);
+      }
+      catch (Exception exc)
+      {
+        errors.Add(errorMessage);
 
-                errors.AddRange(response.Message.Errors);
+        _logger.LogError(exc, errorMessage);
+      }
 
-                _logger.LogWarning("Can not find user data with this id {UserId}: " +
-                    $"{Environment.NewLine}{string.Join('\n', response.Message.Errors)}", userId);
-            }
-            catch (Exception exc)
-            {
-                errors.Add(errorMessage);
+      return null;
+    }
 
-                _logger.LogError(exc, errorMessage);
-            }
+    public FindTasksCommand(
+      ITaskInfoMapper mapper,
+      ITaskRepository taskRepository,
+      IUserRepository userRepository,
+      ILogger<FindTasksCommand> logger,
+      IAccessValidator accessValidator,
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IGetUsersDataRequest> requestClient,
+      IBaseFindRequestValidator findRequestValidator)
+    {
+      _mapper = mapper;
+      _logger = logger;
+      _requestClient = requestClient;
+      _taskRepository = taskRepository;
+      _userRepository = userRepository;
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _findRequestValidator = findRequestValidator;
+    }
 
-            return null;
-        }
+    public FindResultResponse<TaskInfo> Execute(FindTasksFilter filter)
+    {
+      Guid userId = _httpContextAccessor.HttpContext.GetUserId();
+      List<DbProjectUser> projectUsers = _userRepository.Find(userId).ToList();
 
-        public FindTasksCommand(
-            ITaskInfoMapper mapper,
-            ITaskRepository taskRepository,
-            IUserRepository userRepository,
-            ILogger<FindTasksCommand> logger,
-            IAccessValidator accessValidator,
-            IHttpContextAccessor httpContextAccessor,
-            IRequestClient<IGetUsersDataRequest> requestClient)
-        {
-            _mapper = mapper;
-            _logger = logger;
-            _requestClient = requestClient;
-            _taskRepository = taskRepository;
-            _userRepository = userRepository;
-            _accessValidator = accessValidator;
-            _httpContextAccessor = httpContextAccessor;
-        }
+      if (!(projectUsers.Any() || _accessValidator.IsAdmin()))
```

Comment:
Do you need to verify the rights in Find?

Reply:
I ask this question on the call.